hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/InodeTree.java

```diff
@@ -458,11 +458,11 @@ private boolean hasFallbackLink() {
    * there will be root to root mapping. So, root does not represent as
    * internalDir.
    */
-  protected boolean isRootInternalDir() {
+  public boolean isRootInternalDir() {
```

Comment:
Why limit to only these methods? Should the other package-private methods also be made public? why/why not?

Reply:
We need to access just these two functions for our current use case. We can make more functions public is they are needed in the future.

Reply:
I don't think that's the right way to think about it - you make public whatever functionality of InodeTree you'd likely use in the future/would be methods that are useful for InodeTree

Reply:
We checked methods from InodeTree that are being used in ViewFileSystem. resolve() and getHomeDirPrefixValue() are the other two methods which are also being used in ViewFileSystem, besides the constructor of InodeTree. Thus, we also made these two methods public.

Reply:
got it. thanks for updating this!